src/utils/genericUtils.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ *
+ * @param date
+ * @returns string
+ *
+ * This function is used to parse a date in any format into a string.
+ */
+export const parseDate = (date: unknown, includeTime = false): string => {
```

Comment:
I would say it is better to use a package like [date-fns](https://www.npmjs.com/package/date-fns), rather than build our own which is likely to have missed some gotchas. Dates in JavaScript are notorious.